### PR TITLE
Add inline CSS options

### PR DESCRIPTION
**Pull request by eligrey**
## Related Issues

- Links https://transcend.height.app/T-39694

## Security Implications

_[none]_

## System Availability

_[none]_

### CODE REVIEW

File: src/consent-manager.tsx [hunk 1]
```diff
@@ -46,27 +52,34 @@ export const injectConsentManagerApp = async (
...
-      appContainer.appendChild(style);
+        if (EXTERNALIZE_INLINE_CSS) {
+          (style as HTMLLinkElement).rel = 'stylesheet';
+          (style as HTMLLinkElement).href = `data:text/css,${encodeURIComponent(CSS_RESET)}`;
+        }
 
-      // Append UI container to doc to activate style.sheet
-      (document.documentElement || document).append(consentManager);
+        appContainer.appendChild(style);
+        attachToDoc();
```
Comment (eligrey) [inline src/consent-manager.tsx:71]: had to move stuff around to keep this optimal. we want to activate the stylesheet _after_ setup for the external one, and activate it _before_ setup for the inline one

Comment (eligrey) [reply src/consent-manager.tsx:71]: (hence the two orthogonal ifs with a between step instead of an if else)

Comment (kate-kazantseva) [reply src/consent-manager.tsx:71]: just wanted to ask this, makes sense

**Review by kate-kazantseva — DISMISSED** [review]

**Review by kate-kazantseva — DISMISSED** [review]

**Review by kate-kazantseva — APPROVED** [review]
3rd time's a charm ✨

File: src/config.ts [hunk 2]
```diff
@@ -210,3 +211,6 @@ if (CSP_NONCE) {
     delete currentScriptDataset.nonce;
   }
 }
+
+export const ALLOW_INLINE_CSS = inlineCss !== 'off';
+export const EXTERNALIZE_INLINE_CSS = inlineCss === 'data:';
```
Comment (csmccarthy) [inline src/config.ts:216]: could you clarify why we check against `data:` here? it doesnt look like we pass the `inlineCss` value anywhere, so it seems to me to just be an odd enum value

Comment (eligrey) [reply src/config.ts:216]: these are options for this parameter. data: is the name of the externalization option atm to afford for a potential future blob: option etc. to best fit varying CSPs

Comment (eligrey) [reply src/config.ts:216]: airgap.js forwards all of its loadOptions to modules that it loads, so the UI module automatically gets this value

**Pull request merged**